test/functional/test_nic_stat_error_inject.py

```diff
@@ -0,0 +1,209 @@
+    global conf
+    if conf:
+        stop_node()
+    os.environ["PATH"] = old_path
+
+
+def start_node():
+    global conf
+    global ssh
+    global s
+    fake_config = fixtures.FakeConfig()
+    conf = fake_config.get_node_info()
+    conf["compute"]["networks"] = [
+        {
+            "device": "vmxnet3",
+            "id": "mgmt",
+            "mac": "00:60:16:9c:50:6a"
+        }
+    ]
+    conf["compute"]["networks"][0]["port_forward"] = [
+        {
+            "outside": 2222,
+            "inside": 22,
+            "protocal": "tcp"
+        }
+    ]
+    conf["compute"]["storage_backend"] = [
+        {
+            "type": "ahci",
+            "max_drive_per_controller": 6,
+            "drives": [
+                {
+                    "size": 8,
+                    "file": fixtures.image
+                }
+            ]
+        },
+    ]
+
+    node = model.CNode(conf)
+    node.init()
+    node.precheck()
+    node.start()
+    path = os.path.join(node.workspace.get_workspace(), ".monitor")
+    s = UnixSocket(path)
+    s.connect()
+    s.recv()
+
+    payload_enable_qmp = {
+        "execute": "qmp_capabilities"
+    }
+
+    s.send(json.dumps(payload_enable_qmp))
+    s.recv()
+    # wait until system is ready for ssh.
+    ssh = sshclient.SSH(host="127.0.0.1", username="root", password="root", port=2222)
+    ssh.wait_for_host_up()
+
+
+def stop_node():
+    global s
+    global conf
+    s.close()
+    node = model.CNode(conf)
+    node.init()
+    node.stop()
+    node.terminate_workspace()
+    conf = {}
+
+
+def get_interface_name():
+    global inface
+    global ssh
+    with open("ifg_info.txt", "w+") as ff:
+        status, output = ssh.exec_command("ifconfig")
+        ff.write(output)
+        ff.seek(0, 0)
+        for str0 in ff.readlines():
+            if "UP,BROADCAST,RUNNING,MULTICAST" in str0:
+                str1 = str0.split()
+                # in order to delete ":"
+                inface = str1[0][0:-1]
+                break
+        ff.close
+
+
+class nicdata_check:
+    Grxstat = 0
+    Grxerr = 0
+    Gtxstat = 0
+    Gtxerr = 0
+
+    def getnicdata(self):
+        status, output = ssh.exec_command('ifconfig -s {}'.format(inface))
+        with open("ifg_s.txt", "w+") as fd:
+            fd.write(output)
+            fd.seek(0, 0)
+            for line in fd.readlines():
+                print(line)
+                line1 = line.split()
+                if line1[0] == inface:
+                    self.Grxstat = long(line1[2])
+                    self.Grxerr = long(line1[3])
+                    self.Gtxstat = long(line1[6])
+                    self.Gtxerr = long(line1[7])
+            fd.close()
+
+
+class test_nic_error_inject(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        start_node()
+
+    @classmethod
+    def tearDownClass(cls):
+        stop_node()
+
+    def test_status_code_error_inject(self):
+        status_code_error = {
```

Comment:
suggest to add some comments here to remind us what these params stand for.

Reply:
ok